### ISSUE TITLE
SideEffectsSniff with php 8.1 enum

### ISSUE THREAD

**Issue by ArthurKacprzak** [squizlabs/PHP_CodeSniffer#3615]
**Describe the bug**
Functions in a php Enum trigger SideEffectsSniff error

**Code sample**
```php
enum tmp: int
{
    case one = 1;
    case two = 2;

    public function label(): string
    {
        return match ($this) {
            tmp::one => 'one',
            tmp::two => 'two',
        };
    }
}
```

**Expected behavior**
An explanation of why I can't create a function in a php enum.

**Comment by jrfnl** [comment]
@ArthurKacprzak I cannot reproduce this with [PHPCS 3.7.0](https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.7.0), which is the first PHPCS version with syntax support for PHP 8.1.

Please upgrade to PHPCS 3.7.0

**Comment by ArthurKacprzak** [comment]
I use phpinsights with squizlabs/php_codesniffer 3.7.0. 

```
Side effects: (PHP_CodeSniffer\Standards\PSR1\Sniffs\Files\SideEffectsSniff)
  /var/www/tmp.php:1: A file should declare new symbols (classes, functions, constants, etc.) and cause no other side effects, or it should execute logic with side effects, but should not do both. The first symbol is defined on line 23 and the first side effect is on line 8.
```

**Comment by ArthurKacprzak** [comment]
so the problem is probably insight, thank you.

**Comment by gsherwood** [comment]
It's likely that whatever tool you are using is not actually using version 3.7.0 of PHP_CodeSniffer. The error will appear on all PHP_CodeSniffer versions < 3.7.0 but I also can't replicate on 3.7.0.
